Source/AwsCommonRuntimeKit/io/SocketOptions.swift

```diff
@@ -3,60 +3,42 @@
-        get { return self.rawValue.pointee.connect_timeout_ms }
-        set(value) { self.rawValue.pointee.connect_timeout_ms = value }
-    }
-
-    public var keepAlive: Bool {
-        get { return self.rawValue.pointee.keepalive }
-        set(value) { self.rawValue.pointee.keepalive = value }
-    }
-
-    public var keepaliveIntervalSec: UInt16 {
-        get { return self.rawValue.pointee.keep_alive_interval_sec }
-        set(value) { self.rawValue.pointee.keep_alive_interval_sec = value }
-    }
-
-    public var keepaliveMaxFailedProbes: UInt16 {
-        get { return self.rawValue.pointee.keep_alive_max_failed_probes }
-        set(value) { self.rawValue.pointee.keep_alive_max_failed_probes = value }
-    }
-
-    public var keepaliveTimeoutSec: UInt16 {
-        get { return self.rawValue.pointee.keep_alive_timeout_sec }
-        set(value) { self.rawValue.pointee.keep_alive_timeout_sec = value }
-    }
-
-    public var socketDomain: SocketDomain {
-        get { return SocketDomain(rawValue: self.rawValue.pointee.domain) }
-        set(value) { self.rawValue.pointee.domain = value.rawValue }
-    }
-
-    public var socketType: SocketType {
-        get { return SocketType(rawValue: self.rawValue.pointee.type) }
-        set(value) { self.rawValue.pointee.type = value.rawValue }
-    }
-
-    deinit {
-        rawValue.deinitializeAndDeallocate()
+public class SocketOptions: CStruct {
+    public var socketType: SocketType
+    public var socketDomain: SocketDomain
+    public var connectTimeoutMs: UInt32
+    public var keepaliveIntervalSec: UInt16
+    public var keepaliveMaxFailedProbes: UInt16
+    public var keepaliveTimeoutSec: UInt16
+    public var keepAlive: Bool
+
+    public init(socketType: SocketType = .stream,
+                socketDomain: SocketDomain = SocketDomain.ipv4,
+                connectTimeoutMs: UInt32 = 3_000,
+                keepaliveIntervalSec: UInt16 = 0,
+                keepaliveMaxFailedProbes: UInt16 = 0,
+                keepaliveTimeoutSec: UInt16 = 0,
+                keepAlive: Bool = false) {
+        self.socketType = socketType
+        self.socketDomain = socketDomain
+        self.connectTimeoutMs = connectTimeoutMs
+        self.keepaliveIntervalSec = keepaliveIntervalSec
+        self.keepaliveMaxFailedProbes = keepaliveMaxFailedProbes
+        self.keepaliveTimeoutSec = keepaliveTimeoutSec
+        self.keepAlive = keepAlive
+
+    }
+
+    typealias RawType = aws_socket_options
+    func withCStruct<Result>(_ body: (aws_socket_options) -> Result) -> Result {
+        var cSocketOptions = aws_socket_options()
+        cSocketOptions.type = socketType.rawValue
+        cSocketOptions.domain = SocketDomain.ipv4.rawValue
+        cSocketOptions.connect_timeout_ms = connectTimeoutMs
+        cSocketOptions.keep_alive_interval_sec = 0
+        cSocketOptions.keep_alive_timeout_sec = 0
+        cSocketOptions.keep_alive_max_failed_probes = 0
+        cSocketOptions.keepalive = false
```

Comment:
forgot to actually set some values, i.e. instead of 0 this line should be:
`cSocketOptions.keep_alive_interval_sec = keepaliveIntervalSec`

Reply:
Thanks, fixed.